File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/bluetooth/BluetoothLockMapper.java
package com.lattis.ellipse.bluetooth;

import androidx.annotation.NonNull;

import com.lattis.ellipse.data.network.base.AbstractDataMapper;
import com.lattis.ellipse.domain.model.Lock;

import javax.inject.Inject;

import io.lattis.ellipse.sdk.model.BluetoothLock;

/**
 * Created by ssd3 on 4/26/17.
 */

public class BluetoothLockMapper extends AbstractDataMapper<Lock,BluetoothLock> {

    AlertMapper alertMapper;

    @Inject
    BluetoothLockMapper(AlertMapper alertMapper) {
        this.alertMapper=alertMapper;
    }

    @NonNull
    @Override
    public BluetoothLock mapIn(@NonNull Lock lock) {
        BluetoothLock bluetoothLock = new BluetoothLock();
        bluetoothLock.setLockId(lock.getLockId());
        bluetoothLock.setMacId(lock.getMacId());
        bluetoothLock.setMacAddress(lock.getMacAddress());
        bluetoothLock.setName(lock.getName());
        bluetoothLock.setPublicKey(lock.getPublicKey());
        bluetoothLock.setSignedMessage(lock.getSignedMessage());
        bluetoothLock.setUserId(lock.getUserId());
        bluetoothLock.setAutoLockActive(lock.isAutoProximityLock());
        bluetoothLock.setAutoUnLockActive(lock.isAutoProximityUnlock());
        bluetoothLock.setAlertMode(alertMapper.mapIn(lock.getAlertMode()));
        if(lock.getActivityClassName()!=null){
            Class activityClass = null;
            try {
                activityClass = Class.forName(lock.getActivityClassName());
                if(activityClass!=null){
                    bluetoothLock.getAlertMode().forActivity(activityClass);
                }
            } catch (ClassNotFoundException e) {
                e.printStackTrace();
            }
        }
        return bluetoothLock;
    }

    @NonNull
    @Override
    public Lock mapOut(@NonNull BluetoothLock bluetoothLock) {
        Lock lock = new Lock();
        lock.setLockId(bluetoothLock.getLockId());
        lock.setMacAddress(bluetoothLock.getMacAddress());
        lock.setMacId(bluetoothLock.getMacId());
        lock.setName(bluetoothLock.getName());
        lock.setSignedMessage(bluetoothLock.getSignedMessage());
        lock.setPublicKey(bluetoothLock.getPublicKey());
        lock.setUserId(bluetoothLock.getUserId());
        lock.setUserId(bluetoothLock.getUserId());
        lock.setUserId(bluetoothLock.getUserId());
        lock.setAutoProximityLock(bluetoothLock.isAutoLockActive());
        lock.setAutoProximityLock(bluetoothLock.isAutoUnLockActive());
        lock.setAlertMode(alertMapper.mapOut(bluetoothLock.getAlertMode()));
        if(bluetoothLock.getAlertMode()!=null){
            if(bluetoothLock.getAlertMode().getActivity()!=null){
                lock.setActivityClassName( bluetoothLock.getAlertMode().getActivity().getName());
            }
        }
        return lock;
    }
}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/mapper/bluetooth/BluetoothLockMapper.kt
package com.lattis.data.mapper.bluetooth

import com.lattis.data.mapper.AbstractDataMapper
import com.lattis.domain.models.Lock
import io.lattis.ellipse.sdk.model.BluetoothLock
import javax.inject.Inject

class BluetoothLockMapper @Inject internal constructor(var alertMapper: AlertMapper) :
    AbstractDataMapper<Lock?, BluetoothLock?>() {
    override fun mapIn(lock: Lock?): BluetoothLock {
        val bluetoothLock = BluetoothLock()
        bluetoothLock.lockId = lock?.lockId
        bluetoothLock.macId = lock?.macId
        bluetoothLock.macAddress = lock?.macAddress
        bluetoothLock.name = lock?.name
        bluetoothLock.publicKey = lock?.publicKey
        bluetoothLock.signedMessage = lock?.signedMessage
        bluetoothLock.userId = lock?.userId
        bluetoothLock.isAutoLockActive = lock?.isAutoProximityLock
        bluetoothLock.isAutoUnLockActive = lock?.isAutoProximityUnlock
        bluetoothLock.alertMode = alertMapper.mapIn(lock?.alertMode)
        if (lock?.activityClassName != null) {
            var activityClass: Class<*>? = null
            try {
                activityClass = Class.forName(lock?.activityClassName!!)
                if (activityClass != null) {
                    bluetoothLock.alertMode.forActivity(activityClass)
                }
            } catch (e: ClassNotFoundException) {

            }
        }
        return bluetoothLock
    }

    override fun mapOut(bluetoothLock: BluetoothLock?): Lock {
        val lock = Lock()
        lock.lockId = bluetoothLock?.lockId
        lock.macAddress = bluetoothLock?.macAddress
        lock.macId = bluetoothLock?.macId
        lock.name = bluetoothLock?.name
        lock.signedMessage = bluetoothLock?.signedMessage
        lock.publicKey = bluetoothLock?.publicKey
        lock.userId = bluetoothLock?.userId
        lock.userId = bluetoothLock?.userId
        lock.userId = bluetoothLock?.userId
        lock.isAutoProximityLock = bluetoothLock?.isAutoLockActive
        lock.isAutoProximityLock = bluetoothLock?.isAutoUnLockActive
        lock.alertMode = alertMapper.mapOut(bluetoothLock?.alertMode)
        if (bluetoothLock?.alertMode != null) {
            if (bluetoothLock?.alertMode.activity != null) {
                lock.activityClassName = bluetoothLock?.alertMode.activity.name
            }
        }
        return lock
    }

}